### ISSUE TITLE
Support USB HID devices (including keyboard emulation)

### ISSUE THREAD

**Issue by jflamy** [wokwi/wokwi-features#306]
When emulating a USB keyboard, it would be nice to have a monitor of some sort to see the key presses being sent.

Conversely,  a virtual keyboard to send keypresses (and perhaps mouse interaction) would be nice.

**Comment by kartben** [comment]
Or maybe similar to wokwi-gdbserver, perhaps there could be a wokwi-usbipserver which exposes the device's USB port over USB-IP? It would then be reasonably straightforward to mount it on the host machine. See e.g. https://docs.microsoft.com/en-us/windows/wsl/connect-usb

**Comment by jflamy** [comment]
@kartben yes, some form of USB proxy, with the ability to peek what goes in and out.

**Comment by urish** [comment]
Than @kartben! As far as I can tell, usbipd-win does the opposite: it will enable sharing of Windows USB devices with whatever is running in the simulator, in case we implemented USB Host support. 

By we need here the other end of the USB-over-IP connection: use a shared USB device in windows. Do you know if such thing exists (and is maintained)?

**Comment by jflamy** [comment]
Have not thought this fully, but would taking the usb output from the
simulator and feed it (somehow) to a USB over IP gateway work ?
The simulator could perhaps emit json-formatted USB "packets" and the
gateway would convert that  and become the client to a Windows USBIP server.
The other way might work as well (gateway implements server side and feeds
the simulated USB)

https://github.com/cezanne/usbip-win

**Comment by urish** [comment]
Thanks for the link @jflamy. Yes, seems like using USB-over-IP might be a good solution. I changed the title accordingly and opened this request for [voting](https://wokwi.com/features?sort=newest#feature-1128736053)

**Comment by ftab** [comment]
There's also WebUSB API

https://wicg.github.io/webusb/

**Comment by jflamy** [comment]
https://wicg.github.io/webhid/ is interesting.  This would be sufficient to simulate cases where the ESP32 is acting as a keypad, or conversely receiving key presses/mouse

**Comment by urish** [comment]
From my experience with WebHID, a few months ago, it's very limited. It doesn't let you use any HID device, and specifically excludes mice, keyboards, etc. You can learn more [here](https://github.com/robatwilliams/awesome-webhid#good-to-know)
